### PR TITLE
Add clarification for `private protected`

### LINKED ISSUE #5408: InternalsVisibleToAttribute doesn't work for private protected access modifier
**Version Used**: 
Microsoft Visual Studio Community 2019 Version 16.9.0
.NET 5.0

**Steps to Reproduce**:

1. Create two assembly with two classes inside each assembly.

AssemblyA:
```csharp
using System.Runtime.CompilerServices;

[assembly: InternalsVisibleTo("AssemblyB")]

namespace AssemblyA
{
	public class Foo
	{
		internal int Bar { get; set; }
		protected internal int Bat { get; set; }
		private protected int Baz { get; set; }
	}
}
```

AssemblyB:
```csharp
using AssemblyA;

namespace AssemblyB
{
	public class Quux
	{
		public Quux()
		{
			var foo = new Foo();
			foo.Bar = 42; // No error
			foo.Bat = 31415; // No error
			foo.Baz = 666; // Error CS0122  'Foo.Baz' is inaccessible due to its protection level
		}
	}
}
```

2. Get an error when trying to access the Baz property:

> Error CS0122  'Foo.Baz' is inaccessible due to its protection level

**Expected Behavior**:
https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.internalsvisibletoattribute?view=net-5.0#remarks

InternalsVisibleToAttribute works for **private protected** access modifier as it does for **internal** and **protected internal** access modifiers.

**Actual Behavior**:
InternalsVisibleToAttribute doesn't work for **private protected** access modifier, but works for **internal** and **protected internal** access modifiers.

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #5408.

[Internal preview link](https://review.docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.internalsvisibletoattribute?view=net-5.0&branch=pr-en-us-5437).

I reworded the first paragraph of the Remarks section. I added `protected internal` to the list of types/members extended by InternalsVisibleToAttribute. I also added a note about `private protected` per #5408. 

cc @Youssef1313

### CODE REVIEW

File: xml/System.Runtime.CompilerServices/InternalsVisibleToAttribute.xml [hunk 1]
```diff
@@ -46,8 +46,9 @@
     <remarks>
       <format type="text/markdown"><![CDATA[  
   
-## Remarks  
- Ordinarily, types and members with `internal` or `private protected` scope (in C#) and `Friend` and `Private Protected` scope (in Visual Basic) are visible only in the assembly in which they are defined. The <xref:System.Runtime.CompilerServices.InternalsVisibleToAttribute> attribute makes them also visible to the types in a specified assembly, which is known as a friend assembly. This only applies to `internal` (`Friend` in VB) or `private protected`(`Private Protected` in VB) methods only, but not `private` ones.
```
Comment (Youssef1313) [inline xml/System.Runtime.CompilerServices/InternalsVisibleToAttribute.xml:50]: @gewarren I believe `private protected` is  correct.

**Comment by Youssef1313** [comment]
> Fixes dotnet/docs#5408.

This a PR. I think you may have copied the wrong link.

File: xml/System.Runtime.CompilerServices/InternalsVisibleToAttribute.xml (hunk 1 above)
Comment (Youssef1313) [reply xml/System.Runtime.CompilerServices/InternalsVisibleToAttribute.xml:50]: From [this test](https://github.com/dotnet/roslyn/blob/6da1274c9d24c2f90a48290394a951b23617f2a3/src/Compilers/CSharp/Test/Emit/PrivateProtected.cs#L83) in Roslyn, it implies that `private protected` is indeed accessible via IVTs. I haven't tried that though.

**Comment by gewarren** [comment]
> > Fixes [dotnet/docs#5408](https://github.com/dotnet/docs/pull/5408).
> 
> This a PR. I think you may have copied the wrong link.

Oops, that issue is actually in this repo. Thanks.

**Comment by gewarren** [comment]
Ping for review?

File: xml/System.Runtime.CompilerServices/InternalsVisibleToAttribute.xml [hunk 2]
```diff
@@ -46,8 +46,14 @@
...
+ 
+ The <xref:System.Runtime.CompilerServices.InternalsVisibleToAttribute> attribute makes these types and members also visible to the types in a specified assembly, which is known as a friend assembly. This applies only to `internal` (`Friend` in Visual Basic), `protected internal`(`Protected Friend` in Visual Basic), and `private protected` (`Private Protected` in Visual Basic) members, but not `private` ones.
+ 
+ > [!NOTE]
+ > In this case of `private protected` (`Private Protected` in Visual Basic) members, the <xref:System.Runtime.CompilerServices.InternalsVisibleToAttribute> attribute extends accessibility only to types in the specified friend assembly *that derive from the containing class*.
```
Comment (buyaa-n) [inline xml/System.Runtime.CompilerServices/InternalsVisibleToAttribute.xml:56]: NIT: for me this `Note` is not so understandable, could use some improvement, how this change looks?
```suggestion
 > In this case of `private protected` (`Private Protected` in Visual Basic) members, the <xref:System.Runtime.CompilerServices.InternalsVisibleToAttribute> attribute extends accessibility only to types that derives from the *containing class* of the member.
```

**Review by buyaa-n — APPROVED** [review]
Overall LGTM, one NIT comment

**Pull request merged**